bin/plugin/commands/release.js

```diff
@@ -720,21 +678,39 @@ async function releasePlugin( isRC = true ) {
 
 	const packageJsonPath = gitWorkingDirectory + '/package.json';
 	const packageJson = readJSONFile( packageJsonPath );
-	const packageVersion = semver.parse( packageJson.version );
-	const isPackageVersionRC = packageVersion.prerelease.length > 0;
-
-	// Creating the release branch
-	const { version, versionLabel, releaseBranch } = ! isPackageVersionRC
-		? await runReleaseBranchCreationStep(
-				gitWorkingDirectory,
-				abortMessage,
-				isRC
-		  )
-		: await runReleaseBranchCheckoutStep(
-				gitWorkingDirectory,
-				abortMessage,
-				isRC
-		  );
+	const packageVersion = packageJson.version;
+	const parsedPackagedVersion = semver.parse( packageJson.version );
+	const isPackageVersionRC = parsedPackagedVersion.prerelease.length > 0;
+
+	// Are we going to release an RC?
+	if ( isRC ) {
+		// We are releasing an RC.
+		// If packageVersion is stable, then generate new branch and RC1.
+		// If packageVersion is RC, then checkout branch and inc RC.
+		versionInfo = ! isPackageVersionRC
+			? await runReleaseBranchCreationStep(
+					gitWorkingDirectory,
+					abortMessage,
+					getNextMajorVersion( packageVersion ) + '-rc.1'
+			  )
+			: await runReleaseBranchCheckoutStep(
+					gitWorkingDirectory,
+					abortMessage,
+					semver.inc( packageVersion, 'prerelease', 'rc' )
+			  );
+	} else {
+		// We are releasing a stable version.
+		// If packageVersion is stable, then checkout the branch and inc patch.
+		// If packageVersion is RC, then checkout the branch and inc patch, effectively removing the RC.
+		versionInfo = await runReleaseBranchCheckoutStep(
+			gitWorkingDirectory,
+			abortMessage,
+			semver.inc( packageVersion, 'patch' )
+		);
+	}
+
+	const { version, releaseBranch } = versionInfo;
```

Comment:
See https://github.com/WordPress/gutenberg/pull/25971/commits/069f5f8cc1c0e6b3d981ad74744f8b9ee81b1fd1#r505636588 -- let's not obtain `version` from the return value of `runReleaseBranchCreationStep`/`runReleaseBranchCheckoutStep`, but simply set a `version` var inside of our `if`/`else` construct, and use that both as an arg _to_ those functions, _and_ here.

We might consider doing the same for `releaseBranch` -- we might just compute that value in this function now, and pass it to `runReleaseBranchCreationStep`/`runReleaseBranchCheckoutStep`.